src/sentry/snuba/events.py

```diff
@@ -211,18 +212,43 @@ class Columns(Enum):
     )
 
 
+class BaseColumn(object):
+    def get(self):
+        raise NotImplementedError
+
+
+class BuiltInColumn(BaseColumn):
+    def __init__(self, col):
+        self.__col = col
+
+    def get(self, dataset=Dataset.Discover):
+        if dataset == Dataset.Events:
+            return self.__col.value.event_name
+        return self.__col.value.discover_name
+
+
+class TagColumn(BaseColumn):
+    def __init__(self, key):
+        self.__key = key
+
+    def get(self, dataset=Dataset.Discover):
```

Comment:
This interface could be a good long term replacement for how we're currently resolving aliases

Reply:
Yeah. I think we could also use this in the other parts of the query.

Reply:
Well, either that or just have eventstore accept already resolved column names like in the other query args.